### ISSUE TITLE
Cannot share a folder with container in Colima

### ISSUE THREAD

**Issue by claudiocoliveira** [abiosoft/colima#479]
### Description

Hi guys,
I cannot share a file when mounting mysql container.
I am running the following command: 
`docker run --name mysql -v "/Users/myusername/mysql":"/var/lib/mysql":rw -p 3306:3306 -e MYSQL_ROOT_PASSWORD=pass -d mysql:5.7.33`
And I am getting: `chown: changing ownership of '/var/lib/mysql/': Permission denied`

I already checked the `~/.lima/colima/lima.yaml` file and the writable is true:
mounts:
    - location: "~"
      writable: true
    - location: /tmp/colima
      writable: true

If I try run the same command in docker desktop it works fine but with colima no.
Anyone can help me?

### Version

Colima Version: v20.10.18
Lima Version: 0.13.0
Qemu Version: 7.1.0

### Operating System

- [X] macOS Intel
- [ ] macOS M1
- [ ] Linux

### Reproduction Steps

1. docker run --name mysql -v "/Users/myusername/mysql":"/var/lib/mysql":rw -p 3306:3306 -e MYSQL_ROOT_PASSWORD=pass -d mysql:5.7.33

### Expected behaviour

It will be create a shared folder between ~/mysql and /var/lib/mysql to prevent lose my databases.

**Comment by altitudedashboard** [comment]
I am getting a similar error for Oracle XE:

```
mkdir -p ~/oradata

colima start --arch x86_64 --mount-type 9p --mount ~/oradata:w --memory 4 --network-driver slirp oraxe
colima list

docker run -d -p 1521:1521 -e ORACLE_PASSWORD=Oracle2022! -v ~/oradata:/opt/oracle/oradata/:rw --name oraxe gvenzl/oracle-xe

docker logs -f oraxe

CONTAINER: starting up...
CONTAINER: first database startup, initializing...
CONTAINER: uncompressing database data files, please wait...
checkdir error:  cannot create /opt/oracle/oradata/XE
                 Permission denied
                 unable to process XE/.
checkdir error:  cannot create /opt/oracle/oradata/XE
                 Permission denied

```

**Comment by abiosoft** [comment]
@claudiocoliveira using the 9p mount should resolve your issue. But you need to delete the VM and recreate it.
```
colima delete
colima start --mount-type 9p
```

@altitudedashboard do you by chance know the user id the oracle container is running as?

**Comment by claudiocoliveira** [comment]
@abiosoft thanks for your answer.
I run the following commands:

- colima delete
- colima start --mount-type 9p
- docker run --name mysql -v "/Users/username/mysql":"/var/lib/mysql":rw -p 3306:3306 -e MYSQL_ROOT_PASSWORD=pass -d mysql:5.7.33
- docker logs mysql

And the error persiste:
2022-11-15 10:33:41+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.33-1debian10 started.
chown: changing ownership of '/var/lib/mysql/': Operation not permitted

**Comment by claudiocoliveira** [comment]
@abiosoft I don't know if it helps but if I change /Users/username/mysql to /mysql it works fine but it does't create a shared folder mysql in directory /.

**Comment by abiosoft** [comment]
@claudiocoliveira can you try specifying the user? `--user $UID:$UID`, as seen in the "Running as an arbitrary user" section of the [official image docs](https://hub.docker.com/_/mysql)

**Comment by claudiocoliveira** [comment]
@abiosoft really thank you, it works well with --user.

**Comment by altitudedashboard** [comment]
@abiosoft The Oracle XE image https://github.com/gvenzl/oci-oracle-xe/ doesn't work with bind volumes:

```
colima start --arch x86_64 --mount-type 9p --memory 4 --network-driver slirp --network-address oraxe
mkdir ~/XE
docker run -d -p 1521:1521 -e ORACLE_PASSWORD=Oracle2022! --user 54321:54321 -v ~/XE:/opt/oracle/oradata/XE:rw --name oraxe gvenzl/oracle-xe
```

fails with:

```
CONTAINER: starting up...
CONTAINER: first database startup, initializing...
CONTAINER: uncompressing database data files, please wait...
checkdir error:  cannot create /opt/oracle/oradata/XE/XEPDB1
                 Permission denied
                 unable to process XE/XEPDB1/.
checkdir error:  cannot create /opt/oracle/oradata/XE/XEPDB1
                 Permission denied
```

colima is the only way for now for people with Mac M1/M2 to run an Oracle DB.
It would be so nice to have a solution. Perhaps the problem is actually with the docker image? Or am I doing something wrong? Any log files to check?

**Comment by altitudedashboard** [comment]
Yes: 54321 for the oracle user

**Comment by abiosoft** [comment]
> Yes: 54321 for the oracle user

That's not gonna work. You either have to use your user's uid (usually `501` on macOS) or create the directory and make it writable by everyone before starting the container.

```
mkdir ~/XE
chmod -R 777 ~/XE
```

**Comment by altitudedashboard** [comment]
I have tried that, it still fails, also with permissions 777, so I guess the problem is the docker image. I will open an issue with them. Thanks!
